File: modules/web/chromium/patch_native_host.py
#!/usr/bin/env python3
import argparse
import json

parser = argparse.ArgumentParser(
    description="Set allowed_origins in a native messaging host manifest"
)
parser.add_argument("extension_id", help="Chrome extension ID")
parser.add_argument("input", help="Path to the input JSON manifest")
parser.add_argument("output", help="Path to write the patched JSON manifest")
args = parser.parse_args()

with open(args.input) as f:
    manifest = json.load(f)

origin = f"chrome-extension://{args.extension_id}/"
manifest["allowed_origins"] = [origin]

with open(args.output, "w") as f:
    json.dump(manifest, f, indent=2)
    f.write("\n")

File: modules/web/chromium/patch_native_host.lua
#!/usr/bin/env luajit
-- patch_native_host: set allowed_origins in a native messaging host manifest.
local dkjson = require("dkjson")

local extension_id = assert(arg[1], "usage: patch_native_host.lua <extension_id> <input> <output>")
local input = assert(arg[2], "usage: patch_native_host.lua <extension_id> <input> <output>")
local output = assert(arg[3], "usage: patch_native_host.lua <extension_id> <input> <output>")

local f = assert(io.open(input, "r"))
local content = f:read("*a")
f:close()

local manifest = dkjson.decode(content)
manifest.allowed_origins = { "chrome-extension://" .. extension_id .. "/" }

f = assert(io.open(output, "w"))
f:write(dkjson.encode(manifest, { indent = true }))
f:write("\n")
f:close()
